File: php/mysql_connection.php
<?php
// MySQL Connection in PHP

$host = "localhost";
$user = "myuser";
$password = "mypassword";
$database = "mydatabase";

function getMySQLConnection($host, $user, $password, $database) {
    try {
        $conn = new mysqli($host, $user, $password, $database);
        if ($conn->connect_error) {
            throw new Exception("Connection failed: " . $conn->connect_error);
        }
        echo "Connected successfully to MySQL database\n";
        return $conn;
    } catch (Exception $e) {
        echo "Error: " . $e->getMessage() . "\n";
        return null;
    }
}

// Initialize MySQL connection
$conn = getMySQLConnection($host, $user, $password, $database);

if ($conn) {
    try {
        // Creating a table
        $sql = "CREATE TABLE IF NOT EXISTS users (
            id INT AUTO_INCREMENT PRIMARY KEY,
            name VARCHAR(100) NOT NULL,
            age INT NOT NULL,
            city VARCHAR(100)
        )";
        $conn->query($sql);
        echo "Table 'users' created or already exists.\n";

        // Inserting data
        $sql = "INSERT INTO users (name, age, city) VALUES ('Alice', 30, 'New York')";
        $conn->query($sql);
        echo "Record inserted into 'users'.\n";

        // Fetching data
        $sql = "SELECT * FROM users";
        $result = $conn->query($sql);
        if ($result->num_rows > 0) {
            while ($row = $result->fetch_assoc()) {
                print_r($row);
            }
        }

        // Updating data
        $sql = "UPDATE users SET age = 31 WHERE name = 'Alice'";
        $conn->query($sql);
        echo "Record updated.\n";

        // Deleting data
        $sql = "DELETE FROM users WHERE name = 'Alice'";
        $conn->query($sql);
        echo "Record deleted.\n";

    } catch (Exception $e) {
        echo "Error: " . $e->getMessage() . "\n";
    } finally {
        $conn->close();
        echo "MySQL connection closed.\n";
    }
}
?>

File: python/mysql_connection.py
import mysql.connector
from mysql.connector import Error


def get_mysql_connection():
    try:
        conn = mysql.connector.connect(
            host="localhost",
            user="myuser",
            password="mypassword",
            database="mydatabase",
        )
        if conn.is_connected():
            print("Connected to MySQL database")
            return conn
    except Error as e:
        print("Error while connecting to MySQL", e)
        return None


# Initialize the MySQL connection
conn = get_mysql_connection()

if conn:
    try:
        # Example: Creating a new table
        cursor = conn.cursor()
        cursor.execute("""
            CREATE TABLE IF NOT EXISTS users (
                id INT AUTO_INCREMENT PRIMARY KEY,
                name VARCHAR(100),
                age INT,
                city VARCHAR(100)
            )
        """)
        print("Table 'users' created or already exists.")

        # Example: Inserting a record
        cursor.execute(
            """
            INSERT INTO users (name, age, city) VALUES (%s, %s, %s)
        """,
            ("Alice", 30, "New York"),
        )
        conn.commit()
        print("Inserted a new record into 'users'.")

        # Example: Querying records
        cursor.execute("SELECT * FROM users")
        rows = cursor.fetchall()
        print("Records in 'users':")
        for row in rows:
            print(row)

        # Example: Updating a record
        cursor.execute(
            """
            UPDATE users SET age = %s WHERE name = %s
        """,
            (31, "Alice"),
        )
        conn.commit()
        print(f"Updated {cursor.rowcount} record(s).")

        # Example: Deleting a record
        cursor.execute("DELETE FROM users WHERE name = %s", ("Alice",))
        conn.commit()
        print(f"Deleted {cursor.rowcount} record(s).")

    except Error as e:
        print("Error executing MySQL operations:", e)
    finally:
        cursor.close()
        conn.close()
        print("MySQL connection is closed.")
